File: Hash Table/387. First Unique Character in a String/unique.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        hashset = [0] * 26

        for i in range(len(s)):
            hashset[ord(s[i]) - ord("a")] += 1

        for i in range(len(s)):
            if hashset[ord(s[i]) - ord("a")] == 1:
                return i
        return -1

File: 4. Hash Table/387. First Unique Character in a String/unique.cpp
class Solution
{
public:
    int firstUniqChar(string s)
    {
        vector<int> hashset(26, 0);

        for (int i = 0; i < s.size(); i++)
        {
            hashset[s[i] - 'a'] += 1;
        }
        for (int i = 0; i < s.size(); i++)
        {
            if (hashset[s[i] - 'a'] == 1)
            {
                return i;
            }
        }

        return -1;
    }
};